rocky-raft/src/rocky/raft/server/FollowerLogic.java

```diff
@@ -86,9 +86,6 @@ protected Message handleMessage(Message message, ServerContext serverContext) th
                     requestVoteRpcReply = new RequestVoteRpcReply(currentTerm, false);
                 } else {
                     int votedFor = serverContext.getVotedFor();
-                    if (serverContext.getConfig().getServerConfig(votedFor) == null) {
```

Comment:
This was a dirty hack which I thought was required. But now that we reset votedFor correctly, shouldn't be needed anymore.